Source/WebKit/NetworkProcess/mac/NetworkConnectionToWebProcessMac.mm

```diff
@@ -105,11 +113,19 @@
 
     RELEASE_LOG(Process, "Launch Services checkin starting, infoDictionary = %{public}@", (__bridge NSDictionary *)infoDictionary.get());
 
-    Boolean ok = _LSApplicationCheckInProxyPtr()(kLSDefaultSessionID, auditToken.auditToken(), infoDictionary.get(), ^(CFDictionaryRef result, CFErrorRef error) {
+    auto block = makeBlockPtr([protectedThis = Ref { *this }, this, auditToken](CFDictionaryRef result, CFErrorRef error) mutable {
```

Comment:
Have you considered capturing a WeakPtr instead in the lambda? Then we wouldn't need ThreadSafeRefCounted at all.

I just don't want to encourage people to use this object off the main thread since it is really not meant to be.

Reply:
That is a great point, fixed in latest PR.

Thanks for reviewing!